domain/application/state/application.go

```diff
@@ -448,35 +471,45 @@ func (st *ApplicationState) insertUnit(
 	return nil
 }
 
-func (st *ApplicationState) upsertUnit(
-	ctx context.Context, tx *sqlair.TX, toUpdate unitDetails, args application.UpsertUnitArg,
+// UpdateUnit updates the specified application unit, returning an error
+// satisfying [applicationerrors.UnitNotFoundError] if the unit doesn't exist.
+func (st *ApplicationState) UpdateUnit(
+	ctx domain.AtomicContext, appID coreapplication.ID, args application.UpsertUnitArg,
```

Comment:
I guess we will see if the `UpsertUnitArg` shared args type will be acceptable long time. I have no immediate problem with it, really it makes sense to stay the same between both update and insert. Maybe the name could be better, but I'm not concerned until we see a pattern emerge across different domains.